client/my-sites/hosting/cache-card/use-toggle-edge-cache.ts

```diff
@@ -0,0 +1,61 @@
+import {
+	useMutation,
+	UseMutationOptions,
+	useQueryClient,
+	useIsMutating,
+} from '@tanstack/react-query';
+import { useCallback } from 'react';
+import wp from 'calypso/lib/wp';
+import { USE_EDGE_CACHE_QUERY_KEY } from './use-edge-cache';
+
+interface MutationError {
+	code: string;
+	message: string;
+}
+
+export const TOGGLE_EDGE_CACHE_MUTATION_KEY = 'set-edge-site-mutation-key';
+
+export const useToggleEdgeCacheMutation = (
+	siteId: number,
+	options: UseMutationOptions< boolean, MutationError, boolean > = {}
+) => {
+	const queryClient = useQueryClient();
+	const mutation = useMutation< boolean, MutationError, boolean >( {
+		mutationFn: async ( active ) => {
+			return wp.req.post( {
+				path: `/sites/${ siteId }/hosting/edge-cache/active`,
+				apiNamespace: 'wpcom/v2',
+				body: {
+					active: active ? 1 : 0,
+				},
+			} );
+		},
+		mutationKey: [ TOGGLE_EDGE_CACHE_MUTATION_KEY, siteId ],
+		onMutate: async ( active ) => {
+			await queryClient.cancelQueries( [ USE_EDGE_CACHE_QUERY_KEY, siteId ] );
+			const previousData = queryClient.getQueryData( [ USE_EDGE_CACHE_QUERY_KEY, siteId ] );
+			queryClient.setQueryData( [ USE_EDGE_CACHE_QUERY_KEY, siteId ], active );
+			return { previousActive: previousData };
+		},
+		onError( _err, _newActive, context ) {
+			// Revert to previous settings on failure
+			queryClient.setQueryData( [ USE_EDGE_CACHE_QUERY_KEY, siteId ], context?.previousActive );
```

Comment:
CI shows type error related to `context?.previousActive`.

Reply:
Fixed that! Thanks!